index.bs

```diff
@@ -2917,29 +2929,31 @@ To <dfn noexport>get an OS-registration URL from a header list</dfn> given a
 1. If |value| is not a [=string=], return null.
 1. Return the result of running the [=URL parser=] on |value|.
 
+To <dfn>get supported registrars</dfn>:
+
+1. Let |supportedRegistrars| be an [=set/is empty|empty=] [=ordered set=].
+1. If the user agent supports web registrations, [=set/append=] "<code>[=registrar/web=]</code>"
+    to |supportedRegistrars|.
+1. If the user agent supports OS registrations, [=set/append=] "<code>[=registrar/os=]</code>"
+    to |supportedRegistrars|.
+1. Return |supportedRegistrars|.
+
 "<code><dfn>Attribution-Reporting-Support</dfn></code>" is a Dictionary
 Structured Header set on a [=request=] that indicates which registrars, if
 any, the corresponding [=response=] can use. Its values are not specified and
-its <dfn lt="support key">allowed keys</dfn> are:
-
-<dl dfn-for="support key">
-: "<dfn><code>web</code></dfn>"
-:: The user agent supports web registrations.
-: "<dfn><code>os</code></dfn>"
-:: The user agent supports OS registrations.
-
-</dl>
+its allowed keys are the [=registrars=].
 
 Issue: Add links to Structured Header RFC.
 
 To <dfn noexport>set an OS-support header</dfn> given a [=header list=]
 |headers|:
 
+1. Let |supportedRegistrars| be the result of [=getting supported registrars=].
 1. Let |dict| be an [=ordered map=].
-1. If the user agent supports web registrations, [=map/set=]
-    |dict|["<code>[=support key/web=]</code>"] to true.
-1. If the user agent supports OS registrations, [=map/set=]
-    |dict|["<code>[=support key/os=]</code>"] to true.
+1. If |supportedRegistrars| [=set/contains=] "<code>[=registrar/web=]</code>", [=map/set=]
+    |dict|["<code>[=registrar/web=]</code>"] to true.
+1. If |supportedRegistrars| [=set/contains=] "<code>[=registrar/os=]</code>", [=map/set=]
+    |dict|["<code>[=registrar/os=]</code>"] to true.
```

Comment:
To avoid listing each possible registrar explicitly, maybe just iterate over `supportedRegistrars`.

Reply:
Done, thanks.